### PR TITLE
doc:Travis CI note on Build Timeouts

### PR DESCRIPTION
**Pull request by RandyMcMillan**
Related to #17420

Added developer note to minimize issues related to Travis CI accounts.

### CODE REVIEW

File: doc/developer-notes.md [hunk 1]
```diff
@@ -461,6 +461,8 @@ General Bitcoin Core
   - *Explanation*: If the test suite is to be updated for a change, this has to
     be done first.
 
+  - ***Developers using their own Travis CI account*** to pretest their commits should work off of a recent master that is passing. This ensures that your work is based on a passing version and any fails are contained within your commit. ***You may have to restart your builds. See: [Build Timeouts](https://docs.travis-ci.com/user/customizing-the-build#build-timeouts)***
```
Comment (laanwj) [inline doc/developer-notes.md]: Adding `***` around this is overkill, it is not significantly more important than the other recommendations. Please format this like the other items.

**Comment by MarcoFalke** [comment]
Please add this documentation to the existing one in the `.travis.yml`. It is not relevant for anyone not using travis.

**Comment by RandyMcMillan** [comment]
Ok

**Comment by laanwj** [comment]
> Please add this documentation to the existing one in the .travis.yml. It is not relevant for anyone not using travis.

Arguably that particular item in the developer notes "Make sure pull requests pass Travis CI before merging." is about Travis. But ok, better to put the documentation closer to where someone might notice it.

This collides 100% with #17420 now, though. Would be better to review that and suggest improvements there.

**Comment by MarcoFalke** [comment]
> developer notes "Make sure pull requests pass Travis CI before merging."

That should probably be removed then.

**Comment by laanwj** [comment]
Or remove "Travis". it's still important to have the CI passing before merge, to prevent it from failing on the branch. As is happening for Appveyor, yet again.

**Comment by MarcoFalke** [comment]
As with the appveyor failure right now, there is often no way for a contributor to make sure the ci passes. The issue might be unrelated to their changes.

It should probably say that the unit and functional tests should pass locally.

**Comment by RandyMcMillan** [comment]
Thanks for the feed back. I believe a few strategic changes can reduce frustration for members and contributors. I think prompting contributors to rebase (if needed) prior to creating a PR can further streamline this a little. Clearly #17420 should be merged first - I can always add language about rebasing after that.

I also put in a feature request with Travis CI to warn about upstream failures on a topic branch. Maybe they will solve that part of the issue in the future upgrade?

**Pull request closed without merging**